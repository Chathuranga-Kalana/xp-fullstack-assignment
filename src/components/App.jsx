import React from "react";
import {
  Link,
  BrowserRouter as Routers,
  Route,
  Switch
} from "react-router-dom";

import Search from "./search/Search";
import User from "./user/User";
import Readme from "./readme/Readme";
import "./app.css";

class App extends React.Component {
  render() {
    return (
      <Routers>
        <div className="main-app">
          <header className="main-header">
            <h3>
              <Link to="/">xp-fullstack-assignment</Link>
            </h3>
          </header>
          <main className="main-content">
            <Switch>
              <Route exact path="/" component={Search}></Route>
              <Route exact path="/user/:username" component={User}></Route>
              <Route
                exact
                path="/readme/:username/:reponame"
                component={Readme}
              ></Route>
            </Switch>
          </main>
        </div>
      </Routers>
    );
  }
}

export default App;
