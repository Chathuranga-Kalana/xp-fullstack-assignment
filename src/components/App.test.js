import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import App from "./App";

configure({ adapter: new Adapter() });

describe("AppComponent", () => {
  it("should render app component", () => {
    const app_comp = shallow(<App />);
    expect(app_comp.getElements()).toMatchSnapshot();
  });
});
