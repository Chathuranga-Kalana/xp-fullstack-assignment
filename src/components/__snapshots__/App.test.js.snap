// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should render app component 1`] = `
Array [
  <BrowserRouter>
    <div
      className="main-app"
    >
      <header
        className="main-header"
      >
        <h3>
          <ForwardRef
            to="/"
          >
            xp-fullstack-assignment
          </ForwardRef>
        </h3>
      </header>
      <main
        className="main-content"
      >
        <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/user/:username"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/readme/:username/:reponame"
          />
        </Switch>
      </main>
    </div>
  </BrowserRouter>,
]
`;
